packages/core/src/Editor.ts

```diff
@@ -406,6 +406,11 @@ export class Editor extends EventEmitter<EditorEvents> {
     const state = this.state.apply(transaction)
     const selectionHasChanged = !this.state.selection.eq(state.selection)
 
+    this.emit('beforeTransaction', {
+      editor: this,
+      transaction,
+      newState: state,
+    })
     this.view.updateState(state)
     this.emit('transaction', {
```

Comment:
Or maybe, what if we emitted this event *before* updating the view's state. That should allow you to read things before the view's state actually applies

Reply:
The current handlers for `transaction` expect it to already by applied, so this will just introduce the same issue in all framework bindings.